src/base/logging.hh

```diff
@@ -123,19 +126,41 @@ class Logger
      */
     [[noreturn]] void exit_helper() { exit(); ::abort(); }
 
+    void
+    registerExtraLog(std::function<std::string()> cb)
+    {
+        extraLog.emplace_back(std::move(cb));
+    }
+
   protected:
     bool enabled;
 
+    std::string
+    getFormattedExtraLog()
+    {
+        std::stringstream ss;
+        for (auto &cb : extraLog) {
+            std::string str = cb();
+            ss << str;
+            if (str.length() && str.back() != '\n' && str.back() != '\r')
+                ss << std::endl;
+        }
+        return ss.str();
+    }
+
     /** Generates the log message. By default it is sent to cerr. */
     virtual void
```

Comment:
Does this function still need to be virtual?

Reply:
https://github.com/gem5/gem5/blob/stable/src/base/gtest/logging_mock.cc#L39

I find out that GTestLogger overrides the log function. So I change the function back to virtual in the latest commit